push_test.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	smartling "github.com/Smartling/api-sdk-go"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+type request struct {
+	response string
+	code     int
+}
+
+type roundTripFunc func(req *http.Request) *http.Response
+
+func (f roundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
+
+func mockHttpClient(fn roundTripFunc) *http.Client {
+	return &http.Client{
+		Transport: fn,
+	}
+}
+
+func TestPushStopUnauthorized(t *testing.T) {
+	args := getArgs("README.md README.md")
+
+	httpClient := getMockHttpClient([]request{{"{}", 401}})
+
+	mockGlobber(args)
+	defer func() {
+		globFilesLocally = globFilesLocallyFunc
+	}()
+
+	client := getClient(httpClient)
+
+	err := doFilesPush(&client, getConfig(), args)
+
+	assert.EqualError(
+		t,
+		err,
+		"ERROR: unable to upload file \"README.md\"\n"+
+			"└─ failed to upload original file: unable to authenticate: "+
+			"authentication parameters are invalid\n\n"+
+			"Check, that you have enough permissions to upload file to the specified project")
+}
+
+func TestPushContinueFakeError(t *testing.T) {
+	args := getArgs("README.md README.md")
+
+	responses := []request{{`{
+    "response": {
+        "code": "SUCCESS",
+        "data": {
+			"accessToken": "accessToken",
+			"refreshToken": "refreshToken"
+        }
+    }
+}`, 200},
+		{`{
+    "response": {
+        "code": "SUCCESS",
+        "data": {
+        }
+    }
+}`, 401},
+		{`{
+    "response": {
+        "code": "SUCCESS",
+        "data": {
+        }
+    }
+}`, 200},
+	}
+
+	httpClient := getMockHttpClient(responses)
+
+	mockGlobber(args)
+	defer func() {
+		globFilesLocally = globFilesLocallyFunc
+	}()
+
+	client := getClient(httpClient)
+
+	err := doFilesPush(&client, getConfig(), args)
+	assert.EqualError(
+		t,
+		err,
+		"ERROR: failed to upload 1 files\n\nfailed to upload files README.md")
+}
+
+func TestPushStopApiError(t *testing.T) {
+	args := getArgs("README.md README.md")
+
+	responses := []request{{`{
+    "response": {
+        "code": "SUCCESS",
+        "data": {
+			"accessToken": "accessToken",
```

Comment:
Strange `data`. Why not leave it empty or use foo\bar. I mean it draws unnecessary attention.